### PR TITLE
Feat/lw 8188 upgrade serialization to conway era

### PR DESCRIPTION
**Pull request by AngelCastilloB**
# Context

We need to upgrade our serialization classes in order to support the Conway era transactions.

# Proposed Solution

Add the necessary serialization classes to support Conway era.

# Important Changes Introduced

New serialization classes were added

### CODE REVIEW

File: packages/core/src/Cardano/types/Certificate.ts [hunk 1]
```diff
@@ -40,7 +40,7 @@ export interface NewStakeAddressCertificate {
 export interface VoteDelegationCertificate {
   __typename: CertificateType.VoteDelegation;
   stakeKeyHash: Crypto.Ed25519KeyHashHex;
-  drep: DelegateRepresentative;
+  dRep: DelegateRepresentative;
```
Comment (mkazlauskas) [inline packages/core/src/Cardano/types/Certificate.ts:43]: This and some other type changes in `Governance.ts` are technically breaking, please reflect it in commit message

**Review by mkazlauskas — CHANGES_REQUESTED** [review]

File: packages/core/src/Cardano/types/Certificate.ts (hunk 1 above)
Comment (mkazlauskas) [reply packages/core/src/Cardano/types/Certificate.ts:43]: If we intend to squash this commit to the one that created Governance.ts then yeah

File: packages/core/src/Serialization/TransactionBody/ProposalProcedure/InfoAction.ts [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+   */
+  static fromCbor(cbor: HexBlob): InfoAction {
+    const reader = new CborReader(cbor);
+
+    const kind = Number(reader.readUInt());
+
+    if (kind !== GovernanceActionKind.Info)
+      throw new InvalidArgumentError(
+        'cbor',
+        `Expected action kind, expected ${GovernanceActionKind.HardForkInitiation}  but got ${kind}`
```
Comment (mchappell) [inline packages/core/src/Serialization/TransactionBody/ProposalProcedure/InfoAction.ts:43]:
```suggestion
        `Expected action kind, expected ${GovernanceActionKind.Info}  but got ${kind}`
```

File: packages/core/src/Serialization/TransactionBody/ProposalProcedure/NewConstitution.ts [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        'cbor',
+        `Expected an array of ${EMBEDDED_GROUP_SIZE} elements, but got an array of ${length} elements`
+      );
+
+    const kind = Number(reader.readUInt());
+
+    if (kind !== GovernanceActionKind.NewConstitution)
+      throw new InvalidArgumentError(
+        'cbor',
+        `Expected action kind, expected ${GovernanceActionKind.HardForkInitiation}  but got ${kind}`
```
Comment (mchappell) [inline packages/core/src/Serialization/TransactionBody/ProposalProcedure/NewConstitution.ts:72]:
```suggestion
        `Expected action kind, expected ${GovernanceActionKind.NewConstitution}  but got ${kind}`
```

**Review by mchappell — CHANGES_REQUESTED** [review]

**Review by rhyslbw — APPROVED** [review]
This is incredible work @AngelCastilloB :mage: 

I'm happy to see this be merged since this era is pre-production, and the priority for us is early integration. We can perform further reviews once there's test vectors available for gaining confidence in the serialisation.

File: packages/core/src/Cardano/types/Certificate.ts (hunk 1 above)
Comment (rhyslbw) [reply packages/core/src/Cardano/types/Certificate.ts:43]: Could also just cherry-pick this commit onto the other branch then squash it when ready to merge. I agree it's part of the same atomic change

**Pull request closed without merging**